pinot-core/src/test/java/com/linkedin/pinot/core/realtime/segment/ConsumingSegmentAssignmentStrategyTest.java

```diff
@@ -0,0 +1,305 @@
+    } catch (InvalidConfigException e) {
+      Assert.assertTrue(expectException);
+    }
+  }
+
+  /**
+   * Tests various scenarios of how segment assignment will be invoked
+   */
+  @Test
+  public void testAssign() {
+    String tableName = "aTableToTest_REALTIME";
+
+    List<String> newSegments;
+    PartitionAssignment partitionAssignment;
+
+    // empty new segments list
+    newSegments = new ArrayList<>();
+    partitionAssignment = new PartitionAssignment(tableName);
+    verifyAssignment(newSegments, partitionAssignment, false, newSegments);
+
+    // non empty new segments list, empty partition assignment
+    LLCSegmentName llcSegmentName0 = new LLCSegmentName(tableName, 0, 0, System.currentTimeMillis());
+    LLCSegmentName llcSegmentName1 = new LLCSegmentName(tableName, 1, 0, System.currentTimeMillis());
+    LLCSegmentName llcSegmentName2 = new LLCSegmentName(tableName, 2, 0, System.currentTimeMillis());
+    newSegments.add(llcSegmentName0.getSegmentName());
+    newSegments.add(llcSegmentName1.getSegmentName());
+    newSegments.add(llcSegmentName2.getSegmentName());
+    verifyAssignment(newSegments, partitionAssignment, true, newSegments);
+
+    // non empty new segments list, non empty partition assignment, partitions match
+    partitionAssignment.addPartition("0", Lists.newArrayList("s1", "s2"));
+    partitionAssignment.addPartition("1", Lists.newArrayList("s3", "s1"));
+    partitionAssignment.addPartition("2", Lists.newArrayList("s2", "s3"));
+    verifyAssignment(newSegments, partitionAssignment, false, newSegments);
+
+    // partition for a segment missing in partition assignment
+    LLCSegmentName llcSegmentName3 = new LLCSegmentName(tableName, 3, 0, System.currentTimeMillis());
+    newSegments.add(llcSegmentName3.getSegmentName());
+    verifyAssignment(newSegments, partitionAssignment, true, newSegments);
+
+    // extra partitions in partition assignment than needed
+    partitionAssignment.addPartition("3", Lists.newArrayList("s1", "s2"));
+    partitionAssignment.addPartition("4", Lists.newArrayList("s3", "s1"));
+    verifyAssignment(newSegments, partitionAssignment, false, newSegments);
+
+    // non llc segment name
+    List<String> goodSegments = Lists.newArrayList(newSegments);
+    newSegments.add("nonLLCSegmentName");
+    verifyAssignment(newSegments, partitionAssignment, false, goodSegments);
+  }
+
+  /**
+   * Tests a segment lifecycle
+   * 1) new CONSUMING segments in a newly created table, residing on consuming servers
+   * 2) consuming segments become ONLINE on the consuming servers and a new sequence of CONSUMING segments is added
+   * 3) ONLINE segments move to completed servers, cCONSUMING segments still on consuming servers
+   * 4) latest CONSUMING segments become OFFLINE
+   * In all these scenarios, we want to check that the final assignment received from the strategy, adheres strictly to what the latest segments say
+   *
+   */
+  @Test
+  public void testSegmentLifecycle() {
```

Comment:
you can add throw Exception here to the test, and that way you don't have to catch all exceptions that you don't expect. Test will fail if exceptions are thrown.